docs/arch/adr-0001-user-preferences-storage.md

```diff
@@ -0,0 +1,57 @@
+
+AWS has integrated Cognito with S3 bucket access controls, as described [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_s3_cognito-bucket.html). Also, there is a JavaScript library, [aws-amplify](https://github.com/aws-amplify/amplify-js), we could use for front-end integration. The library has Cognito authentication support and also a storage component that supports and defaults to using S3. The library appears to be popular and currently maintained. It supports both other authentication frameworks and storage backends, so should the project have need to switch either, it may be possible to do so with minimal code changes.
+
+
+### No authentication
+
+The client-provided wireframes only show the user entering the voucher holder name and voucher number. If the voucher number is treated like a password, it could be used to access, for example, an S3 file of the user preferences that uses the voucher number as part of the path or file name.
+
+This approach would be the simplest, but would not protect the user preferences from access by other parties who might have the name and voucher number. Access could be restricted by IP address to only allow users in the counseling office; however, this would prevent use of the app directly by the clients, outside of the office.
+
+A potential benefit of not having user accounts is that there would be no need to distinguish between counselor accounts, which need to be able to access any voucher holder's stored preferences, and user accounts, which should have access restricted to only their own stored preferences. Potential downsides, in addition to the lack of access control, would include the inability to audit access, or to track access by user for purposes of gathering analytic data.
+
+
+## Decision
+
+We will use AWS Cognito for authentication to control access to user preference files stored on S3. This approach will protect user data, enable counselor access, and best support future expansion of the project by establishing the means for data access control and auditing as well as the potential for user analytics. Also, as we were already planning to use AWS products for other aspects of this project, using AWS for authentication as well should simplify our stack and ease integration of user management.
+
+
+## Consequences
+
+The consequences of this decision will be:
+
+ - User account management must be set up, adding complexity
```

Comment:
Unclear how much "account management" is attempting to encapsulate here, but I think there could be some challenges managing the individual account lifecycle (creating, editing, deleting) _and_ getting the various Cognito resources (user pools, user groups, IAM polices, email templates, etc.) setup across environments.

A handful of Terraform resources exist now (they didn't when we setup PGW), so it might be feasible to do without [manual interactions in the console](https://github.com/azavea/pgw-construction-map/tree/develop/doc/cognito-setup), but we haven't tried it yet. I anticipate that we'll need to use Terraform for this project to setup the prerequisite resources needed by `mastarm`.